tcp-ballerina/tests/mock_server.bal

```diff
@@ -32,20 +32,15 @@ service on echoServer {
 
     isolated remote function onConnect(Caller caller) returns ConnectionService {
         io:println("Client connected to echoServer: ", caller.remotePort);
-        return new EchoService(caller);
+        return new EchoService();
     }
 }
 
 service class EchoService {
-    Caller caller;
 
-    public isolated function init(Caller c) {
-        self.caller = c;
-    }
-
-    remote function onBytes(readonly & byte[] data) returns (readonly & byte[])|Error? {
+    remote function onBytes(Caller caller, readonly & byte[] data) returns (readonly & byte[])|Error? {
```

Comment:
In this case can't we change the signature as below ?

```ballerina
remote function onBytes(Caller caller, readonly & byte[] data) 
```

Reply:
we can

Reply:
ballerina```
    remote function onBytes(Caller caller, readonly & byte[] data) returns Error? {
        io:println("Echo: ", 'string:fromBytes(data));
        Error? res = caller->writeBytes(data);
    }
    ``` it's possible this way

Reply:
updated to `remote function onBytes(Caller caller, readonly & byte[] data) returns Error?`